kernel/generic/src/ipc/kbox.c

```diff
@@ -249,13 +250,15 @@ int ipc_connect_kbox(task_id_t taskid)
 	
 	if (task->kb.thread != NULL) {
 		mutex_unlock(&task->kb.cleanup_lock);
-		return phone_handle;
+		*out_phone = phone_handle;
+		return EOK;
 	}
 	
 	/* Create a kbox thread */
 	thread_t *kb_thread = thread_create(kbox_thread_proc, NULL, task,
 	    THREAD_FLAG_NONE, "kbox");
 	if (!kb_thread) {
+		// FIXME: Shouldn't we clean up phone_handle?
```

Comment:
With the present code the phone would be cleanup up during ipc_cleanup() when the task is destroyed. If you want to clean-up sooner, which may be a good thing, you could either mimic steps from phone_cap_cleanup_cb() (but there might be some calls sent over it already) or you could swap the ordering of kbox thread creation and phone allocation so that the phone gets allocated when the function cannot fail for other reasons.